docker-compose/7.2.N-docker-compose.yml

```diff
@@ -119,6 +131,12 @@ services:
         -Dalfresco.port=8080
         -Dalfresco.context=alfresco
         -Dalfresco.protocol=http
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.share.rule=PathPrefix(`/share`)"
+      - "traefik.http.services.share.loadbalancer.server.port=8080"
+      - "traefik.http.routers.proxiedsolrapideny.rule=PathRegexp(`^/share/proxy/alfresco(-(noauth|feed|api))?/api/solr/`)"
+      - "traefik.http.routers.proxiedsolrapideny.middlewares=fakeauth@docker"
```

Comment:
Middleware defined in different services can be applied anywhere?

Reply:
so it seems so though that's strange. Before I disabled the `exposedByDefault` feature of the provisioner I had all middlewares defined at the proxy level and I was just referencing them in the services which needed them (and proxy service was quite at the end of the compose file). After disabling it, traefik started to behave as if the declarations had to be set in the of the file. So to answer your question: Yes you can declare a middleware in some service and reuse later (but not before) with `exposedByDefault` set to `false`. At least that's what I witnessed 🤷